src/resolvers/any.rs

```diff
@@ -0,0 +1,95 @@
+// RGB smart contracts for Bitcoin & Lightning
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Written in 2024 by
+//     Zoe Faltibà <zoefaltiba@gmail.com>
+//
+// Copyright (C) 2024 LNP/BP Standards Association. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rgbstd::resolvers::ResolveHeight;
+use rgbstd::validation::{ResolveWitness, WitnessResolverError};
+use rgbstd::{WitnessAnchor, WitnessId, XAnchor, XPubWitness};
+
+#[cfg(feature = "electrum")]
+use crate::electrum;
+#[cfg(feature = "esplora_blocking")]
+use crate::esplora_blocking;
+
+/// Type that contains any of the [`Resolver`] types defined by the library
+pub enum AnyResolver {
+    #[cfg(feature = "electrum")]
+    /// Electrum resolver
+    Electrum(electrum::Resolver),
+    #[cfg(feature = "esplora_blocking")]
+    /// Esplora resolver
+    Esplora(esplora_blocking::Resolver),
+}
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Display, Error, From)]
+#[display(doc_comments)]
+pub enum AnyAnchorResolverError {
+    #[cfg(feature = "electrum")]
+    #[from]
+    #[display(inner)]
+    Electrum(electrum::AnchorResolverError),
+    #[cfg(feature = "esplora_blocking")]
+    #[from]
+    #[display(inner)]
+    Esplora(esplora_blocking::AnchorResolverError),
+}
+
+macro_rules! impl_from {
+    ( $from:ty, $to:ty, $variant:ident, $( $cfg:tt )* ) => {
+        $( $cfg )*
+        impl From<$from> for $to {
+            fn from(inner: $from) -> Self {
+                <$to>::$variant(inner)
+            }
+        }
+    };
+}
+
+macro_rules! impl_inner_method {
+    ( $enum_name:ident, $self:expr, $name:ident $(, $args:expr)* ) => {
+        match $self {
+            #[cfg(feature = "electrum")]
+            $enum_name::Electrum(inner) => inner.$name( $($args, )* ).map_err(|e| e.into()),
+            #[cfg(feature = "esplora_blocking")]
+            $enum_name::Esplora(inner) => inner.$name( $($args, )* ).map_err(|e| e.into()),
+        }
+    }
+}
+
+impl_from!(electrum::Resolver, AnyResolver, Electrum, #[cfg(feature = "electrum")]);
+impl_from!(esplora_blocking::Resolver, AnyResolver, Esplora, #[cfg(feature = "esplora_blocking")]);
+
+impl ResolveHeight for AnyResolver {
+    type Error = AnyAnchorResolverError;
+
+    fn resolve_anchor(&mut self, anchor: &XAnchor) -> Result<WitnessAnchor, Self::Error> {
+        impl_inner_method!(AnyResolver, self, resolve_anchor, anchor)
```

Comment:
While you can do it, in practice I prefer to avoid macros for generating code since it makes code longer to compile, harder to audit and statically analyze. I mean here we save just 3 lines and add a lot of human-reader untransparency